### PR TITLE
Randomly throw signal events containing taus

### PR DESCRIPTION
**Pull request by blinkseb**
The HH signals used for Moriond 17 have bugs in the generation. The BR
of taus decays is incorrect (= BR(tau -> all) instead of BR(tau -> e / mu))

Mitigate the issue by throwing randomly some events containing taus to
retrieve BR(tau -> e / mu)

I also updated the GTs to use the latest JECs (note: https://github.com/cp3-llbb/Datasets/ **must** also be updated)

### CODE REVIEW

**Comment by OlivierBondu** [comment]
For the record, it's not a 'bug' in the sense the pythia fragment is doing exactly what we agreed it should be doing.... as we did not want to have hadronic taus in the sample since we wanted to maximize the 'useful' stat of the sample... and we are now exactly reverting this decision, which is what we need for a quick fix.

but ok, it's semantics at this stage, thanks !

**Comment by vidalm** [comment]
No, we wanted the leptonic decays in the right proportion... (If is doable and how, it is another thing)

**Comment by OlivierBondu** [comment]
Which is possible, but I don't think a pythia filter is what you need / want... Analysis-level recipe, which is what we discussed earlier, is _still possible_, but *not* a quick fix as this one is.

I will take the blame as needed, but guys, I did ask every single one of you about this...

**Comment by swertz** [comment]
It's not a bug, it's a feature, and if there is blame to be taken it's for all of us, since as you say we gave our :+1: when the samples were submitted!

**Comment by vidalm** [comment]
First, I am not blaming you. And I will never blame you for doing something I am not able to do. I thank you for that. 
Second, the only point here is that we were not smart enough (at least I wasn't) to see the outcome of the filter. But I am not saying is your fault.

File: plugins/HHAnalyzer.cc [hunk 1]
```diff
@@ -29,9 +29,277 @@ void HHAnalyzer::registerCategories(CategoryManager& manager, const edm::Paramet
...
+            } else if (pdg_id == -5) {
+                ASSIGN_HH_GEN_INFO(Bbar, "Bbar");
+            }
+
+            // Ignore B decays
+            if (pruned_decays_from_pdg_id(ip, 5, false))
+                continue;
+
+            // Count the number of tau coming directly from a W or a Z
+            if ((pdg_id == 15) && (pruned_decays_from_pdg_id(ip, 24, true) || pruned_decays_from_pdg_id(ip, 23, true))) {
```
Comment (swertz) [inline plugins/HHAnalyzer.cc:191]: Shouldn't this be `abs(pdg_id)`?

Comment (blinkseb) [reply plugins/HHAnalyzer.cc:191]: Indeed

**Pull request merged**